src/Servers/IIS/IIS/src/Core/IISHttpContextSafeHandle.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.IIS.Core;
+
+internal class IISHttpContextSafeHandle : NativeSafeHandle
+{
+    public override bool IsInvalid => handle == IntPtr.Zero;
+
+    public IISHttpContextSafeHandle(IntPtr handle) : base(handle)
+    {
+    }
+
+    internal IISHttpContext? Context { get; set; }
+
+    internal NativeMethods.REQUEST_NOTIFICATION_STATUS Status { get; set; }
+
+    protected override void Dispose(bool disposing)
+    {
+        NativeMethods.HttpSetManagedRequestComplete(this);
+
+        base.Dispose(disposing);
+    }
+
+    protected override bool ReleaseHandle()
+    {
+        // Dispose the context
+        Context!.Dispose();
```

Comment:
If https://github.com/dotnet/aspnetcore/pull/46176 is merged, this Dispose call needs to run after `HttpIndicateCompletion`. Also, shouldn't `handle` below be disposed instead of relying on the finalizer?

Reply:
This isn't a finalizer

Reply:
The handle on line 40